## Question 1
#### Part a

In [3]:
import numpy as np
import seaborn as sns
import pandas as pd

In [10]:
adultdf = pd.read_csv('adult.data')

In [11]:
adultdf.shape

(32560, 15)

In [12]:
adultdf.head()

39          State-gov   77516   Bachelors   13        Never-married  \
0  50   Self-emp-not-inc   83311   Bachelors   13   Married-civ-spouse   
1  38            Private  215646     HS-grad    9             Divorced   
2  53            Private  234721        11th    7   Married-civ-spouse   
3  28            Private  338409   Bachelors   13   Married-civ-spouse   
4  37            Private  284582     Masters   14   Married-civ-spouse   

         Adm-clerical   Not-in-family   White     Male   2174   0   40  \
0     Exec-managerial         Husband   White     Male      0   0   13   
1   Handlers-cleaners   Not-in-family   White     Male      0   0   40   
2   Handlers-cleaners         Husband   Black     Male      0   0   40   
3      Prof-specialty            Wife   Black   Female      0   0   40   
4     Exec-managerial            Wife   White   Female      0   0   40   

    United-States   <=50K  
0   United-States   <=50K  
1   United-States   <=50K  
2   United-States   <=50K  
3            Cuba   <=50K  
4   United-States   <=50K

- There are 32560 datapoints in the dataset
- There does not seem to be a row denoting the columns, I believe that the first row is being used as the columns instead of actual column titles

In [18]:
adultdf.columns

Index(['39', ' State-gov', ' 77516', ' Bachelors', ' 13', ' Never-married',
       ' Adm-clerical', ' Not-in-family', ' White', ' Male', ' 2174', ' 0',
       ' 40', ' United-States', ' <=50K'],
      dtype='object')

In [21]:
adultdf[' Male'].describe()

count     32560
unique        2
top        Male
freq      21789
Name:  Male, dtype: object

- The title of ' Male' leads with a space I had many difficulties figuring this issue out.

#### Part b

In [23]:
adult = pd.read_csv('adult.csv')

In [24]:
adult.shape

(32561, 15)

- There are 32561 Adults, it does confirm my suspicision from part a

In [25]:
adult.columns

Index(['age', 'workclass', 'fnlwgt', 'education', 'education-num',
       'marital-status', 'occupation', 'relationship', 'race', 'sex',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'salary'],
      dtype='object')

In [26]:
adult['sex'].value_counts()

sex
Male      21790
Female    10771
Name: count, dtype: int64

- There are 21790 Men and 10771 Women

In [29]:
adult.groupby('sex').describe()

age                                                       fnlwgt  \
          count       mean        std   min   25%   50%   75%   max    count   
sex                                                                            
Female  10771.0  36.858230  14.013697  17.0  25.0  35.0  46.0  90.0  10771.0   
Male    21790.0  39.433547  13.370630  17.0  29.0  38.0  48.0  90.0  21790.0   

                       ... capital-loss         hours-per-week             \
                 mean  ...          75%     max          count       mean   
sex                    ...                                                  
Female  185746.311206  ...          0.0  4356.0        10771.0  36.410361   
Male    191771.449013  ...          0.0  3770.0        21790.0  42.428086   

                                                
              std  min   25%   50%   75%   max  
sex                                             
Female  11.811300  1.0  30.0  40.0  40.0  99.0  
Male    12.119755  1.0  40.0  40.0  49.0  99.0  

[2 rows x 48 columns]

- The average age for a Female in this dataset is roughly 36.86 years old

In [30]:
adult.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [31]:
adult['native-country'].value_counts()

native-country
United-States                 29170
Mexico                          643
?                               583
Philippines                     198
Germany                         137
Canada                          121
Puerto-Rico                     114
El-Salvador                     106
India                           100
Cuba                             95
England                          90
Jamaica                          81
South                            80
China                            75
Italy                            73
Dominican-Republic               70
Vietnam                          67
Guatemala                        64
Japan                            62
Poland                           60
Columbia                         59
Taiwan                           51
Haiti                            44
Iran                             43
Portugal                         37
Nicaragua                        34
Peru                             31
France       

In [32]:
adult['native-country'].value_counts().sum()

32561

In [34]:
print(137/32561) #Proportion of Germany citizens to all

0.004207487485028101


In [36]:
adult.groupby('salary').describe()

age                                                       fnlwgt  \
          count       mean        std   min   25%   50%   75%   max    count   
salary                                                                         
<=50K   24720.0  36.783738  14.020088  17.0  25.0  34.0  46.0  90.0  24720.0   
>50K     7841.0  44.249841  10.519028  19.0  36.0  44.0  51.0  90.0   7841.0   

                      ... capital-loss         hours-per-week             \
                mean  ...          75%     max          count       mean   
salary                ...                                                  
<=50K   190340.86517  ...          0.0  4356.0        24720.0  38.840210   
>50K    188005.00000  ...          0.0  3683.0         7841.0  45.473026   

                                                
              std  min   25%   50%   75%   max  
salary                                          
<=50K   12.318995  1.0  35.0  40.0  40.0  99.0  
>50K    11.012971  1.0  40.0  40.0  50.0  99.0  

[2 rows x 48 columns]

- For less than 50k it is roughly 36.78 year mean with a 14 year std
- For more than 50k it is roughly 44.25 year mean with a 10.5 year std

In [45]:
adult[(adult['salary'] == '>50K')]['education'].value_counts()

education
Bachelors       2221
HS-grad         1675
Some-college    1387
Masters          959
Prof-school      423
Assoc-voc        361
Doctorate        306
Assoc-acdm       265
10th              62
11th              60
7th-8th           40
12th              33
9th               27
5th-6th           16
1st-4th            6
Name: count, dtype: int64

- It is not true, that those making above 50K all have at least high school education, there is 89 adults without any highschool

In [50]:
adult.groupby(['race','sex'])['age'].agg('max')

race                sex   
Amer-Indian-Eskimo  Female    80
                    Male      82
Asian-Pac-Islander  Female    75
                    Male      90
Black               Female    90
                    Male      90
Other               Female    74
                    Male      77
White               Female    90
                    Male      90
Name: age, dtype: int64

- Its 82 of the oldest Male Amer-Indian_Eskimo for the dataset